src/handshake/server.rs

```diff
@@ -214,12 +235,26 @@ impl<'a, T: AsyncRead + AsyncWrite + Unpin> Server<'a, T> {
     }
 }
 
+fn header_to_str(bytes: &[u8]) -> &str {
+    str::from_utf8(bytes).unwrap_or("INVALID_UTF8")
```

Comment:
That doesn't seem very attack-proof?

Reply:
`Origin` headers are not attack-proof by definition since they are headers, you can set them to anything you want outside of a browser.